File: ucare_frontend/src/service/boardService.js
import axios from 'axios';
import { BOARD_API_BASE_URL } from './urlConfig'

class medicineService {
  create(data) {
    return axios.post(BOARD_API_BASE_URL + '/create', data);
  }
  retrieveAll() {
    return axios.get(BOARD_API_BASE_URL + '/retrieveAll');
  }
  retrieveContents(boardNo) {
    return axios.get(BOARD_API_BASE_URL + '/retrieveContents' + boardNo);
  }
  delete(boardNo) {
    return axios.delete(BOARD_API_BASE_URL + '/delete/' + boardNo);
  }
  update(data) {
    return axios.put(BOARD_API_BASE_URL + '/update', data)
  }
  updateHit(boardNo) {
    return axios.put(BOARD_API_BASE_URL + '/updateHit/' + boardNo)
  }
}

export default new medicineService();

File: ucare_backend/src/main/java/com/douzone/ucare/service/BoardService.java
package com.douzone.ucare.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.douzone.ucare.repository.BoardRepository;
import com.douzone.ucare.vo.BoardVo;

@Service
public class BoardService {
	@Autowired
	private BoardRepository boardRepository;
	
	public int create(BoardVo data) {
		return boardRepository.create(data);
	}
	
	public List<BoardVo> retrieveAll() {
		return boardRepository.retrieveAll();
	}

	public List<BoardVo> retrieveContents(Long boardNo) {
		return boardRepository.retrieveContents(boardNo);
	}

	public int updateHit(Long boardNo) {
		return boardRepository.updateHit(boardNo);
	}
	
	public int delete(Long boardNo) {
	return boardRepository.delete(boardNo);
	}

	public int update(BoardVo data) {
	return boardRepository.update(data);
	}

}
